### DISCUSSION TITLE
[Bug]: Generation fails when configuring a margin with high values

### DISCUSSION THREAD

**Discussion by lairg99** [spatie/laravel-pdf#58] [Ideas]
### What happened?

When I set `margins(300, 200, 200, 200, 'cm')` the following error messages will be produced. Without margin everything works fine.

```The command "PATH=$PATH:/usr/local/bin:/opt/homebrew/bin NODE_PATH=`npm root -g` node '/Users/sam/code/laravel-pdf/vendor/spatie/browsershot/src/../bin/browser.cjs' '{"url":"file:\/\/\/var\/tmp\/1185728813-0979418001706015855\/index.html","action":"pdf","options":{"args":[],"viewport":{"width":800,"height":600},"displayHeaderFooter":false,"margin":{"top":"300cm","right":"200cm","bottom":"200cm","left":"200cm"},"printBackground":true}}'" failed. Exit Code: 1(General error) Working directory: /Users/sam/code/laravel-pdf/public Output: ================ Error Output: ================ [1m[43m[30m Puppeteer old Headless deprecation warning:[0m[33m In the near future `headless: true` will default to the new Headless mode for Chrome instead of the old Headless implementation. For more information, please see https://developer.chrome.com/articles/new-headless/. Consider opting in early by passing `headless: "new"` to `puppeteer.launch()` If you encounter any bugs, please report them to https://github.com/puppeteer/puppeteer/issues/new/choose.[0m ProtocolError: Protocol error (Page.printToPDF): invalid print parameters: content area is empty at <instance_members_initializer> (/Users/sam/code/laravel-pdf/node_modules/puppeteer-core/lib/cjs/puppeteer/common/CallbackRegistry.js:86:14) at new Callback (/Users/sam/code/laravel-pdf/node_modules/puppeteer-core/lib/cjs/puppeteer/common/CallbackRegistry.js:90:16) at CallbackRegistry.create (/Users/sam/code/laravel-pdf/node_modules/puppeteer-core/lib/cjs/puppeteer/common/CallbackRegistry.js:22:26) at Connection._rawSend (/Users/sam/code/laravel-pdf/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/Connection.js:80:26) at CdpCDPSession.send (/Users/sam/code/laravel-pdf/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/CDPSession.js:66:33) at CdpPage.createPDFStream (/Users/sam/code/laravel-pdf/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/Page.js:823:63) at CdpPage.pdf (/Users/sam/code/laravel-pdf/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/Page.js:850:37) at getOutput (/Users/sam/code/laravel-pdf/vendor/spatie/browsershot/bin/browser.cjs:50:43) at callChrome (/Users/sam/code/laravel-pdf/vendor/spatie/browsershot/bin/browser.cjs:385:27) at process.processTicksAndRejections (node:internal/process/task_queues:95:5)```

### How to reproduce the bug

- Puppeteer freshly installed, also reinstalled
- Laravel 10 freshly installed
- node v21.2.0
- macOS 14.2.1

```
class InvoiceController extends Controller
{
    public function __invoke() {
        return pdf()
            ->view('pdf.invoice')
            ->margins(300, 200, 200, 200, 'cm'); // <- without margin it works
    }
}
```

```
// invoice.blade.php
Test
```

### Package Version

1.1.2

### PHP Version

8.2.12

### Laravel Version

10.41.0

**Comment by freekmurze** [comment]
This is probably caused by the high values you pass. Try with lower values of a smaller metric.

**Comment by lairg99** [comment]
Thanks for the quick response. You're right. But shouldn't' it be the same output?

```
->margins(300, 200, 200, 200, 'mm') // <- doesn't work
->margins(3, 2, 2, 2, 'cm') // <- works
```

In my first example, I wrote `cm` by mistake. Of course I meant `mm`.

**Comment by freekmurze** [comment]
It seems that this is caused by something inside of Chromium. I'd appreciate a PR that throw an exception earlier when too big values are passed.

**Reply by lairg99** [reply]
I'll submit a PR this week.

**Comment by lairg99** [comment]
That's embarrassing! 200 mm are not 2cm. 🤦‍♂️ However, an exception should be thrown when the margin is greater than the width. I thought it would be a one-liner, but different metrics need to be considered. I will take a look.
